assets/src/components/activities/multi_input/MultiInputDelivery.tsx

```diff
@@ -120,6 +120,19 @@ export const MultiInputComponent: React.FC = () => {
     {},
   );
 
+  const inputValue = (input: MultiInput) => {
+    const studentInput = uiState.partState[input.partId].studentInput
+
+    // safeguard against null/undefined studentInput
+    if (!studentInput) return '';
+
+    if (input.inputType === 'text') {
+      return uiState.partState[input.partId].studentInput.join(' ');
```

Comment:
So it seems like for 'text' inputs, student input was being stored as an array of strings, one element for each word?

If there is even a chance at some point in time "some" inputs were being stored correctly as full strings, but then an implementation changed which caused them to be stored as an array - we should make this impl here more robust and perhaps be looking to see what `typeof  uiState.partState[input.partId].studentInput` returns.  If it is a string, we use it as is, if it is an array, we `join`

Reply:
I don't think it was ever stored as an array. I believe it was always stored as a string. The bug arised because a client side code uses a utility function safelySelectInputs to fetch the input values from the state. That utility function splits string-valued inputs into arrays before returning. 

It seems use of safelySelectInputs  caused a bug for ShortAnswers two years ago which was fixed by adding a different function safelySelectInputStrings which does not do any splitting. I believe multi-input should use safelySelectinputStrings as well.